### PR TITLE
Add usage comment in hack/update-gofmt.sh

### PR DESCRIPTION
**Pull request by KobayashiD27**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design

/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Add usage comment in hack/update-gofmt.sh.

I cannot find the README of scripts in hack/.
If this explanation is useful, I also add an explanation to another scripts.

**Which issue(s) this PR fixes**:

Ref #86597 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/update-gofmt.sh [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 # GoFmt apparently is changing @ head...
 
+# This script runs GoFmt to golang source codes
```
Comment (tanjunchen) [inline hack/update-gofmt.sh]: GoFmt
advice:
`go fmt`

Comment (KobayashiD27) [reply hack/update-gofmt.sh]: Thanks, your advice seems right.
I will fix it.

**Review by fejta — COMMENTED** [review]
Please bundle this into a more substantial PR (can you do all the hack/*.sh scripts in one go, or a modest subset of them)?

I do not want to review 50 3 line PRs like this.

**Pull request closed without merging**